test/test_optim.py

```diff
@@ -2680,5 +2680,25 @@ def test_bn_update_eval_momentum(self):
 
 instantiate_parametrized_tests(TestLRScheduler)
 
+def _diff_fn(p, grad, opt_differentiable_state, opt_class, kwargs):
+    p = p.clone()
+    p.grad = grad
+    opt_differentiable_state = {k: v.clone() for k, v in opt_differentiable_state.items()}
+    opt = opt_class([p], **kwargs)
+    opt.state.update(opt_differentiable_state)
+    opt.step()
+    return (p,) + tuple(opt_differentiable_state.values())
+
+
+class TestDifferentiableOptimizer(TestCase):
+
+    def test_sgd(self):
+        p = torch.rand(10, requires_grad=True, dtype=torch.float64)
+        grad = torch.rand(10, requires_grad=True, dtype=torch.float64)
+        mbuff = torch.rand(10, requires_grad=True, dtype=torch.float64)
+        state = {'momentum_buffer': mbuff}
+        gradcheck(_diff_fn, (p, grad, state, torch.optim.SGD, {'lr': 0.9, 'differentiable': True}))
```

Comment:
gradcheck will not "unpack" data structures so the content of `state` will not be checked here.
It is fine with the inputs being captured though, so the simplest change you can do here is add a new argument to `_diff_fn` that is `*ignored`. And pass in `, *state.values())` as part of the arguments.

Reply:
Yep exactly!